File: HomeTask2Java/src/Mac.java
public class Mac extends Computer implements ISleep{
    public Mac(){}
    public Mac(String macAddress){
        super(macAddress);
    }

    @Override
    public void turnOn() {
        System.out.println("Mac turn on!");
    }

    @Override
    public void turnOff() {
        System.out.println("Mac turn off!");
    }

    @Override
    public void sleep() {
        System.out.println("Mac in sleep mode!");
    }

    @Override
    public String toString(){
        return String.format("\t" + getClass() + "\n\tmacAddress = " + getMacAddress());
    }
}

File: HomeTask2/HomeTask2/Mac.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HomeTask2
{
    public class Mac : Computer, ISleep
    {
        public Mac(string macAddress) : base(macAddress){}

        public void Sleep()
        {
            Console.WriteLine("Mac in sleep mode!");
        }

        public override void TurnOff()
        {
            Console.WriteLine("Mac turn off!");
        }

        public override void TurnOn()
        {
            Console.WriteLine("Mac turn on!");
        }
        public override string ToString()
        {
            return String.Format("{0}\n\t macAddress = {1}", GetType(), MacAddress);
        }
    }
}
